packages/flutter/lib/src/painting/notched_shapes.dart

```diff
@@ -111,3 +118,44 @@ class CircularNotchedRectangle implements NotchedShape {
       ..close();
   }
 }
+
+/// A [NotchedShape] created from [ShapeBorder]s.
+///
+/// Two shapes can be provided. The [host] is the shape of the widget that
+/// uses the [NotchedShape] (typically a [BottomAppBar]). The [guest] is
+/// subtracted from the [host] to create the notch (typically to make room
+/// for a [FloatingActionButton]).
+class AutomaticNotchedShape extends NotchedShape {
+  /// Creates a [NotchedShape] that is defined by two [ShapeBorder]s.
+  ///
+  /// The [host] must not be null.
+  ///
+  /// The [guest] may be null, in which case no notch is created even
+  /// if a guest rectangle is provided to [getOuterPath].
+  const AutomaticNotchedShape(this.host, [ this.guest ]);
+
+  /// The shape of the widget that uses the [NotchedShape] (typically a
+  /// [BottomAppBar]).
+  ///
+  /// This shape cannot depend on the [TextDirection], as no text direction
+  /// is available to [NotchedShape]s.
+  final ShapeBorder host;
+
+  /// The shape to subtract from the [host] to make the notch.
+  ///
+  /// This shape cannot depend on the [TextDirection], as no text direction
+  /// is available to [NotchedShape]s.
+  ///
+  /// If this is null, [getOuterPath] ignores the guest rectangle.
+  final ShapeBorder guest;
+
+  @override
+  Path getOuterPath(Rect hostRect, Rect guestRect) { // ignore: avoid_renaming_method_parameters
```

Comment:
Why should we add this `// ignore` tech debt? Can we just rename the params in the super-class? If not, can  you add a comment justifying why we need to take on this tech debt (e.g. linking to an analyzer issue or some other explanation)?

Reply:
I can rename the superclass parameters if you like. It'll push the lint to any implementers of this class. It's not clear to me that `hostRect` is a better name than `host` in general. It's only a better name here because it clashes with the property names otherwise (and makes the method really freaking confusing).

Reply:
I'll add a comment, per the new section in our style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#comment-all-code-ignores-code